src/machine/machine_nrf.go

```diff
@@ -234,8 +234,8 @@ func (i2c I2C) readByte() byte {
 func (i2c I2C) readLastByte() byte {
 	for i2c.Bus.EVENTS_RXDREADY == 0 {
 	}
-	i2c.Bus.EVENTS_RXDREADY = 0
 	i2c.signalStop() // signal 'stop' now, so it is sent when reading RXD
+	i2c.Bus.EVENTS_RXDREADY = 0
```

Comment:
What if you remove this line entirely from here?

Reply:
No trying that then it doesn't work at all.